### PR TITLE
SKIP-523 remove need for docker pull

### PR DESCRIPTION
**Pull request by odasko**
## Background
GCP Binary Authorization requires the _image_url_ to be in `registry/repository@digest` format. 
For the user, supplying the _image_url_ in `registry/repository@digest` format is more difficult than supplying it in `registry/repository:tag` format.
For these reasons we required the user to supply the _image_url_ in `registry/repository:tag` format and then acquired the `registry/repository@digest` format by doing `docker pull image_url` and ...

## Issue
Pulling the image is neither time nor resource effective, since the image could be really large.

## Solution 
As suggested by [Github Support](https://support.github.com/ticket/personal/0/1760462) we use the [`docker manifest inspect`](https://docs.docker.com/engine/reference/commandline/manifest_inspect/) command with `--verbose` option to get an image digest without pulling it. 
Additionally we allow the user to provide the _image_url_ in both `registry/repository:tag` and `registry/repository@digest` format 

### Tested?
Tested in https://github.com/kartverket/github-runner/pull/46 (see related Actions)

### CODE REVIEW

**Review by Flexman94 — APPROVED** [review]

File: .github/workflows/post-build-attest.yml [hunk 1]
```diff
@@ -30,11 +33,27 @@ jobs:
...
+      - name: Get Image Digest URL from Image Tag URL
+        id: image-digest-url
+        run: |
+          # If image_url = registry/repository@digest
+          if [[ "${{inputs.image_url}}" == *"@"* ]] 
+          then
+            msg="${{ inputs.image_url }}"
+          else
+            # Convert registry/repository:tag to registry/repository@digest
+            a=${{ inputs.image_url }}
```
Comment (aa-kv) [inline .github/workflows/post-build-attest.yml:46]: Variable name `a` isn't descriptive

File: .github/workflows/post-build-attest.yml [hunk 2]
```diff
@@ -30,11 +33,27 @@ jobs:
...
+            mysha=$(docker manifest inspect $a -v | jq -r .Descriptor.digest)
+            b=${a%:*}
```
Comment (aa-kv) [inline .github/workflows/post-build-attest.yml:48]: Variable name `b` isn't descriptive

File: .github/workflows/post-build-attest.yml [hunk 3]
```diff
@@ -75,7 +94,7 @@ jobs:
       - name: Check if already attested
         id: get-attestation
         run: |
-          mystr=$(gcloud container binauthz attestations list --project="kubernetes-dev-94b9" --attestor="projects/kubernetes-dev-94b9/attestors/Build-Master-Branch" --artifact-url="${{ inputs.image_url }}")
+          mystr=$(gcloud container binauthz attestations list --project="kubernetes-dev-94b9" --attestor="projects/kubernetes-dev-94b9/attestors/Build-Master-Branch" --artifact-url="${{ needs.attest-kartverket-kontekst.outputs.image_url }}")
```
Comment (aa-kv) [inline .github/workflows/post-build-attest.yml:97]: Variable name `mystr` isn't descriptive

File: .github/workflows/run-terraform.yml [hunk 4]
```diff
@@ -408,9 +408,17 @@ jobs:
...
-          docker pull ${{ inputs.image_url}}
-          mystr=$(docker inspect --format='{{index .RepoDigests 0}}' ${{ inputs.image_url}})
-          msg=$(printf '%s' $mystr | jq --raw-input --slurp '.' --join-output)
+          # If image_url = registry/repository@digest
+          if [[ "${{inputs.image_url}}" == *"@"* ]] 
+          then
+            msg="${{ inputs.image_url }}"
+          else
+            # Convert registry/repository:tag to registry/repository@digest
+            a=${{ inputs.image_url }}
```
Comment (aa-kv) [inline .github/workflows/run-terraform.yml:417]: Variable name `a` isn't descriptive

File: .github/workflows/run-terraform.yml [hunk 5]
```diff
@@ -408,9 +408,17 @@ jobs:
...
+            mysha=$(docker manifest inspect $a -v | jq -r .Descriptor.digest)
+            b=${a%:*}
```
Comment (aa-kv) [inline .github/workflows/run-terraform.yml:419]: Variable name `b` isn't descriptive

**Review by aa-kv — APPROVED** [review]

File: .github/workflows/post-build-attest.yml (hunk 1 above)
Comment (odasko) [reply .github/workflows/post-build-attest.yml]: fikset

**Pull request merged**